src/as_execution/abi.rs

```diff
@@ -260,20 +260,21 @@ pub(crate) fn assembly_script_create_sc(
     Ok(StringPtr::alloc(&address, env.get_ffi_env(), &mut ctx)?.offset() as i32)
 }
 
-/// performs a hash on a string and returns the bs58check encoded hash
+/// performs a hash on a bytearray and returns the hash
 #[named]
-pub(crate) fn assembly_script_hash(mut ctx: FunctionEnvMut<ASEnv>, value: i32) -> ABIResult<i32> {
+pub(crate) fn assembly_script_hash(mut ctx: FunctionEnvMut<ASEnv>, bytes: i32) -> ABIResult<i32> {
```

Comment:
As we implemented `assembly_script_hash_sha256`may be it's time to create `assembly_script_hash_black3` ? instead of refactoring this ABI

Reply:
I'm not a big fan of changing the naming as this endpoint give a massa hash and will always give a massa hash without caring what's inside.
Also regarding the time we have and the work we need to do before the end of month this will require gas costs changes etc and so I don't thing we will have time or it will be rush.

Reply:
I see, we could see `_hash` for Massa Hash which could evolve in the future.

Otherwise, `_hash_THE_NAME_OF_HASHING_ALGO`